utils/calendrical_calculations/src/chinese_based.rs

```diff
@@ -264,42 +272,78 @@ pub(crate) fn midnight<C: ChineseBased>(moment: Moment) -> Moment {
 /// Determines the fixed date of the lunar new year given the start of its corresponding solar year (歲), which is
 /// also the winter solstice
 ///
+/// The return values are:
+///
+/// - The calculated lunar new year
+/// - The date of the following solstice
+/// - The number of days between `prior_solstice` and December 20 if `prior_solstice` is too early
+///
+/// The third argument, called `drift_delta`, must be subtracted from dates when checking their
+/// major solar term.
+///
 /// Based on functions from _Calendrical Calculations_ by Reingold & Dershowitz.
 /// Lisp reference code: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L5370-L5394
-pub(crate) fn new_year_in_sui<C: ChineseBased>(prior_solstice: RataDie) -> (RataDie, RataDie) {
+pub(crate) fn new_year_in_sui<C: ChineseBased>(
+    prior_solstice: RataDie,
+) -> (RataDie, RataDie, DriftDelta) {
     // s1 is prior_solstice
     // Using 370 here since solstices are ~365 days apart
-    // Both solstices should fall in December
+    // Both solstices should fall in December, and the first solstice MUST fall on
+    // December 20, 21, or 22. It doesn't seem to change the outcome if the following
+    // solstice is out of range.
+    let (prior_solstice, drift_delta) = {
+        let adjusted = bind_winter_solstice::<C>(prior_solstice);
+        (adjusted, DriftDelta(adjusted - prior_solstice))
+    };
     let following_solstice = winter_solstice_on_or_before::<C>(prior_solstice + 370); // s2
-    debug_assert_eq!(
```

Comment:
issue: keep these asserts for modern year ranges (±2000 years?)

Reply:
The asserts are obsolete if I use the `bind` functions.